storage/cache/log_subtree_cache.go

```diff
@@ -74,13 +74,13 @@ func populateLogSubtreeNodes(hasher hashers.LogHasher) storage.PopulateSubtreeFu
 					return nil
 				}
 
-				subDepth := logStrataDepth - height
-				nodeID := storage.NewNodeIDFromPrefix(st.Prefix, subDepth, index, logStrataDepth, maxLogDepth)
-				_, sfx := nodeID.Split(len(st.Prefix), int(st.Depth))
-				sfxKey := sfx.String()
 				// Don't put leaves into the internal map and only update if we're rebuilding internal
 				// nodes. If the subtree was saved with internal nodes then we don't touch the map.
 				if height > 0 && len(st.Leaves) == maxLeaves {
+					subDepth := logStrataDepth - height
+					nodeID := storage.NewNodeIDFromPrefix(st.Prefix, subDepth, index, logStrataDepth, maxLogDepth)
```

Comment:
_Might_ be worth seeing if there's a cheaper approach to figuring out the internal node suffix?

`NewNodeIDFromPrefix` looks quite heavyweight, and we don't care about the prefix here.

Reply:
Yep, was trying to limit the frightfulness factor for the moment. TODO it?

Reply:
I think we should micro benchmark this code before more serious optimizations.

Reply:
Sure :)

Reply:
I suspect the cost of hashing dominates but it would be good to know.

Reply:
TODO added.